StudentDetails.java

```diff
@@ -0,0 +1,155 @@
+    }
+
+    public String getSlname() {
+        return slname;
+    }
+
+    public String getSfname() {
+        return sfname;
+    }
+
+    public String toString() {
+        return sno + " " + sfname + " " + slname;
+    }
+}
+
+class StudentDetails {
+    private static final int ADD_STUDENT = 1;
+    private static final int SHOW_STUDENT = 2;
+    private static final int SEARCH_STUDENT = 3;
+    private static final int REMOVE_STUDENT = 4;
+    private static final int UPDATE_STUDENT = 5;
+
+    public static void main(String[] args) {
+
+        List<StudentData> c = new ArrayList<StudentData>();
+        Scanner obj = new Scanner(System.in);
+        Scanner obj1 = new Scanner(System.in);
+        int UserChoice;
+        do {
+            System.out.println("Available menu items: ");
+            System.out.println("1: Add student");
+            System.out.println("2: Show student");
+            System.out.println("3: Search student");
+            System.out.println("4: Remove student");
+            System.out.println("5: Update student");
+            System.out.println("which function do you want to excute? ");
+            System.out.print("Menu item: ");
+            UserChoice = obj.nextInt();
+
+            switch (UserChoice) {
+                case ADD_STUDENT: 
+                    System.out.println("Please provide the data for the student to store. ");
+                    System.out.print("Student id: ");
+                    int sno = obj.nextInt();
+                    System.out.print("First name: ");
+                    String sfname = obj1.nextLine();
+                    System.out.print("Last name: ");
+                    String slname = obj1.nextLine();
+                    c.add(new StudentData(sno, sfname, slname));
+                    break;
+                case SHOW_STUDENT:
+                    System.out.println("--------------------------------------------------");
+                    System.out.printf("%-10s %-15s %-15s", "id", "firstName", "LastName");
+                    System.out.println();
+                    Iterator<StudentData> i = c.iterator();
+                    while (i.hasNext()) {
+                        StudentData e = i.next();
+                        System.out.println();
+                        System.out.printf("%-10s %-15s %-15s", e.getSno(), e.getSfname(), e.getSlname());
+
+                    }
+                    System.out.println();
+                    System.out.println("--------------------------------------------------");
+                    break;
+                case SEARCH_STUDENT:
+                    boolean found = false;
+                    System.out.println("Please give the student id to search? ");
+                    System.out.print("Student id:");
+                    int sno1 = obj.nextInt();
+                    System.out.println("--------------------------------------------------");
+                    i = c.iterator();
```

Comment:
Instead of using an Iterator, you can also just use a normal for each loop.
for(StudentData currentStudent : c) { ... }
You can also use stream and filter like described in the Moodle page -> https://rit-moodle.inso.tuwien.ac.at/mod/page/view.php?id=662